server/src/test/java/org/elasticsearch/index/mapper/KeyedJsonFieldTypeTests.java

```diff
@@ -90,6 +120,26 @@ public void testFuzzyQuery() {
         assertEquals("[fuzzy] queries are not currently supported on [json] fields.", e.getMessage());
     }
 
+    public void testRangeQuery() {
+        KeyedJsonFieldType ft = createDefaultFieldType();
+        ft.setName("field");
+
+        TermRangeQuery expected = new TermRangeQuery("field",
+            new BytesRef("key\0lower"),
+            new BytesRef("key\0upper"), false, false);
+        assertEquals(expected, ft.rangeQuery("lower", "upper", false, false, null));
```

Comment:
should we also check that we do the right thing if the bounds are inclusive?

Reply:
👍